### PR TITLE
DataCollector check with only Uri in Runsettings

### PR DESCRIPTION
**Pull request by hvinett**
## Description
removing friendly name check when only Uri is given in run settings
If Uri is correct it proceeds with Uri otherwise if friendly name is correct we get the first uri with that friendly name and proceeds with it. In rest of the cases it shows error.

## Related issue
#1952 

## Behavior
| Friendly name  |   Uri | Result | 
|---------|-------|--------|
|correct |  null  |  Find URI using friendly name |
|correct |  empty/invalid |  Invalid URI Exception |
|correct |  wrong uri  |   Find URI using friendly name  |
|null |  correct|  Use this URI |
|empty |  correct  |   Invalid Friendly Name Exception |
|wrong  |  correct |  Use the URI |
|correct |  correct |  Use the URI directly |
|wrong |  wrong |  Invalid data collector warning |

### CODE REVIEW

File: src/Microsoft.TestPlatform.CrossPlatEngine/DataCollection/DataCollectionLauncherFactory.cs [hunk 1]
```diff
@@ -28,7 +28,11 @@ internal static IDataCollectionLauncher GetDataCollectorLauncher(IProcessHelper
                 var dataCollectionRunSettings = XmlRunSettingsUtilities.GetDataCollectionRunSettings(settingsXml);
                 foreach (var dataCollectorSettings in dataCollectionRunSettings.DataCollectorSettingsList)
                 {
-                    if (dataCollectorSettings.FriendlyName.ToLower().Equals("event log"))
+                    if (dataCollectorSettings.FriendlyName!=null && dataCollectorSettings.FriendlyName.ToLower().Equals("event log"))
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.CrossPlatEngine/DataCollection/DataCollectionLauncherFactory.cs]: Merge these two if conditions.

File: src/Microsoft.TestPlatform.Common/DataCollection/DataCollectionManager.cs [hunk 2]
```diff
@@ -422,7 +422,12 @@ private void LoadAndInitialize(DataCollectorSettings dataCollectorSettings, stri
                 // Look up the extension and initialize it if one is found.
                 var extensionManager = this.DataCollectorExtensionManager;
                 var dataCollectorUri = string.Empty;
-                this.TryGetUriFromFriendlyName(dataCollectorSettings.FriendlyName, out dataCollectorUri);
+
+                if (!this.TryGetUriFromFriendlyName(dataCollectorSettings.FriendlyName, out dataCollectorUri))
+                {
+                    this.LogWarning(string.Format(CultureInfo.CurrentUICulture, Resources.Resources.CannotFetchByFriendlyName, dataCollectorSettings.FriendlyName));
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.Common/DataCollection/DataCollectionManager.cs]: I dont think the error message is correct.

File: src/Microsoft.TestPlatform.Common/DataCollection/DataCollectionManager.cs [hunk 3]
```diff
@@ -422,7 +422,12 @@ private void LoadAndInitialize(DataCollectorSettings dataCollectorSettings, stri
                 // Look up the extension and initialize it if one is found.
                 var extensionManager = this.DataCollectorExtensionManager;
                 var dataCollectorUri = string.Empty;
-                this.TryGetUriFromFriendlyName(dataCollectorSettings.FriendlyName, out dataCollectorUri);
+
+                if (!this.TryGetUriFromFriendlyName(dataCollectorSettings.FriendlyName, out dataCollectorUri))
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.Common/DataCollection/DataCollectionManager.cs]: Shouldn't we do this when datacollectorUri is not specified by the user ?
If user already specified it, and that's what we need, why are trying to fetch it again ?

File: src/Microsoft.TestPlatform.Common/Resources/xlf/Resources.cs.xlf [hunk 4]
```diff
@@ -269,6 +269,11 @@
         <target state="translated">Operace se ruší na základě žádosti.</target>
         <note />
       </trans-unit>
+      <trans-unit id="CannotFetchByFriendlyName">
+        <source>Cannot Fetch Uri with {0}.</source>
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.Common/Resources/xlf/Resources.cs.xlf]: Unable to find a datacollector with friendly name '[0}'

**Comment by singhsarab** [comment]
@hvinett Can you please resolve the comments and complete this one first.

File: src/Microsoft.TestPlatform.Common/Resources/xlf/Resources.cs.xlf (hunk 4 above)
Comment (hvinett) [reply src/Microsoft.TestPlatform.Common/Resources/xlf/Resources.cs.xlf]: changed as mentioned

File: src/Microsoft.TestPlatform.Common/DataCollection/DataCollectionManager.cs (hunk 2 above)
Comment (hvinett) [reply src/Microsoft.TestPlatform.Common/DataCollection/DataCollectionManager.cs]: changed as mentioned

File: src/Microsoft.TestPlatform.Common/DataCollection/DataCollectionManager.cs [hunk 5]
```diff
@@ -391,6 +391,19 @@ protected virtual bool TryGetUriFromFriendlyName(string friendlyName, out string
             return false;
         }
 
+        protected virtual bool IsUriValid(string uri)
+        {
+            var extensionManager = this.dataCollectorExtensionManager;
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.Common/DataCollection/DataCollectionManager.cs]: Null check

**Review by singhsarab — APPROVED** [review]

**Pull request merged**